google/cloud/aiplatform/datasets.py

```diff
@@ -118,11 +120,13 @@ def create(
                 is defined as an OpenAPI 3.0.2 Schema Object. The schema files
                 that can be used here are found in gs://google-cloud-
                 aiplatform/schema/dataset/metadata/.
-            source: Optional[Sequence[str]]=None:
+            gcs_source: Optional[Sequence[str]]=None:
```

Comment:
`gcs_source` seems to be an array or URI's while `bq_source` seems to be a single URI. 

Combining them into one parameter seems messy (even if both were the same datatype) so I split them.
I can combine them into a single `source` parameter again once I refactor it to use a single interface.

Reply:
Also, there's specific logic that changes depending on `bq_source` and `gcs_source` which I feel so really be abstracted out to decouple Dataset from GCS and BQ, as well as allow future extensibility. I'll put a PR together later for discussion.